testsuite/tests/typing-modal-kinds/everything.ml

```diff
@@ -0,0 +1,291 @@
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod contended
+         because of the definition of bad at line 1, characters 0-44.
+|}]
+
+type good : float64 mod contended = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(*** yielding ***)
+type bad : value mod unyielding = t_value
+[%%expect{|
+Line 1, characters 0-41:
+1 | type bad : value mod unyielding = t_value
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_value" is value
+         because of the definition of t_value at line 1, characters 0-20.
+       But the kind of type "t_value" must be a subkind of value mod unyielding
+         because of the definition of bad at line 1, characters 0-41.
+|}]
+
+type good : value mod unyielding = t_value_mod_e
+[%%expect{|
+type good = t_value_mod_e
+|}]
+
+type bad : float64 mod unyielding = t_float64
+[%%expect{|
+Line 1, characters 0-45:
+1 | type bad : float64 mod unyielding = t_float64
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod unyielding
+         because of the definition of bad at line 1, characters 0-45.
+|}]
+
+type good : float64 mod unyielding = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(*** external ***)
+type bad : value mod external_ = t_value
+[%%expect{|
+Line 1, characters 0-40:
+1 | type bad : value mod external_ = t_value
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_value" is value
+         because of the definition of t_value at line 1, characters 0-20.
+       But the kind of type "t_value" must be a subkind of value mod external_
+         because of the definition of bad at line 1, characters 0-40.
+|}]
+
+type good : value mod external_ = t_value_mod_e
+[%%expect{|
+type good = t_value_mod_e
+|}]
+
+type bad : float64 mod external_ = t_float64
+[%%expect{|
+Line 1, characters 0-44:
+1 | type bad : float64 mod external_ = t_float64
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod external_
+         because of the definition of bad at line 1, characters 0-44.
+|}]
+
+type good : float64 mod external_ = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(* mod everything does not affect nullability *)
+type t : value_or_null mod everything
+type bad : value = t
+
+[%%expect{|
+type t
+  : value_or_null mod global aliased many contended portable unyielding
+                      external_
```

Comment:
I think we should try to print `mod everything` here too

Reply:
Sorry, I meant to write a comment about printing: my understanding is that printing jkinds remains a minefield that is due for a substantial rework.  I propose putting off any thought about how to print this well until that rework, I suspect it's not trivial.  Can we compromise on a CR?

Reply:
It's less of a minefield now than it was. I think detecting the everything case and using `everything` there would be easy and valuable. I do think we'll revisit this, but this bit of fruit seems low enough to pluck now.

Reply:
OK, I realized the dumbest possible thing works: The reason this one doesn't print as `value_or_null mod everything` is just that that is not one of the kinds in the list `Builtin.all`.  And you might think we can't add all the possible layout/nullability pairs to that list - they are infinite.  But because of how hilariously broken printing for product kinds is (it pushes the modal bounds down to the base layouts), actually it's enough to just enumerate the base ones.

So, I've added the remaining few missing valid base layout/nullability pairs to the list, and now it works.